public/modules/custom/helfi_hakuvahti/tests/src/Kernel/HakuvahtiInstallTest.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\helfi_hakuvahti\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Tests for Hakuvahti install and update hooks.
+ *
+ * @group helfi_hakuvahti
+ */
+class HakuvahtiInstallTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'helfi_hakuvahti',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->installConfig(['helfi_hakuvahti']);
+
+    // Load install file once for all tests.
+    require_once dirname(__DIR__, 3) . '/helfi_hakuvahti.install';
+  }
+
+  /**
+   * Tests that hook_install() creates default config with PROJECT_NAME.
+   */
+  public function testInstallCreatesDefaultConfig(): void {
+    // Set PROJECT_NAME environment variable.
+    putenv('PROJECT_NAME=test-project');
+
+    // Run hook_install().
+    helfi_hakuvahti_install();
```

Comment:
This seems a bit convoluted to me. I think what you want to test is that correct config entity is created when the module is installed. For that, you should actually install the module with: `$this->container->get('module_installer')->install(['helfi_hakuvahti']);` instead of running the `hook_install` by hand.